src/org/labkey/test/params/FieldDefinition.java

```diff
@@ -160,6 +166,56 @@ public Integer getScale()
         return _scale;
     }
 
+    public Optional<Boolean> isHidden()
+    {
+        return _hidden;
+    }
+    public FieldDefinition isHidden(Boolean hidden)
+    {
+        _hidden = Optional.of(hidden);
+        return this;
+    }
+
+    public Optional<Boolean> isPrimaryKey()
+    {
+        return  _isPrimaryKey;
+    }
+    public FieldDefinition isPrimaryKey(Boolean isPrimaryKey)
+    {
+        _isPrimaryKey = Optional.of(isPrimaryKey);
+        return this;
+    }
+
+    public Optional<Boolean> shownInDetailsView()
+    {
+        return _shownInDetailsView;
+    }
+    public FieldDefinition shownInDetailsView(boolean showInDetailsView)
+    {
+        _shownInDetailsView = Optional.of(showInDetailsView);
+        return this;
+    }
+
+    public Optional<Boolean> shownInInsertView()
+    {
+        return _shownInInsertView;
+    }
+    public FieldDefinition shownInInsertView(boolean showInDetailsView)
+    {
+        _shownInInsertView = Optional.of(showInDetailsView);
+        return this;
+    }
+
+    public Optional<Boolean> shownInUpdateView()
+    {
+        return _shownInUpdateView;
+    }
+    public FieldDefinition shownInUpdateView(boolean showInUpdateView)
+    {
+        _shownInUpdateView = Optional.of(showInUpdateView);
+        return this;
+    }
```

Comment:
Use standard setter/getter method names for all these new fields. (e.g. `setHidden`, `isShownInDetailsView`, `setShownInDetailsView`)
IntelliJ has a feature to generate these methods for you: https://www.jetbrains.com/help/idea/generating-code.html#generate-getters-setters
You can even tell it to generate builder-style setters like we have here